flycheck.el

```diff
@@ -4071,7 +4097,12 @@ Return a list with the contents of the table cell."
          (msg-and-checker (flycheck-error-list-make-last-column flushed-msg checker))
          (explainer (flycheck-checker-get checker 'error-explainer)))
     (list error
-          (vector (flycheck-error-list-make-number-cell
+          (vector (flycheck-error-list-make-cell
+                   (if filename
+                       (file-name-nondirectory filename)
+                     "")
```

Comment:
I think this reduces to `(file-name-nondirectory (or filename ""))`… not sure it reads better.

Reply:
It has the same effect and is more concise, but it calls `file-name-nondirectory` unnecessarily on `""`, so I'd say keep the `if`. (This came from the original PR, and I didn't find a better way to express it.)